spec/controllers/orders_controller_spec.rb

```diff
@@ -69,97 +53,368 @@
+    end
+
+    it "assigns @order" do
+      get :receipt
+
+      expect(assigns(:order)).to eq(checked_out_order)
+    end
+
+    it "is a success" do
       get :receipt
 
-      expect(assigns(:order)).to eq(Order.find(session[:order_id]))
-    end
-
-    # it "assigns @order_items" do
-    #   get :receipt
-    #
-    #   expect(assigns(:order_items)).to eq(OrderItem.all)
-    #   expect(assigns(:order_items_count)).to eq(@order_items_count)
-    # end
-
-    # context "redirecting to other places based on order status" do
-    #   it "redirects to checkout page when order status is pending" do
-    #     Order.find(session[:order_id]).update(status: "pending")
-    #     Order.find(session[:order_id]).save
-    #     Order.find(session[:order_id]).reload
-    #
-    #     get :receipt
-    #
-    #     expect(response).to have_http_status(302)
-    #     expect(response).to redirect_to(checkout_path)
-    #   end
-    #
-    #   it "redirects to root when order status is complete" do
-    #     Order.find(session[:order_id]).update(status: "cancelled")
-    #     Order.find(session[:order_id]).save
-    #     Order.find(session[:order_id]).reload
-    #
-    #     get :receipt
-    #
-    #     expect(response).to have_http_status(302)
-    #     expect(response).to redirect_to(root_path)
-    #   end
-    #
-    #   it "redirects to root when order status is cancelled" do
-    #     Order.find(session[:order_id]).update(status: "complete")
-    #     Order.find(session[:order_id]).save
-    #     Order.find(session[:order_id]).reload
-    #
-    #     get :receipt
-    #
-    #     expect(response).to have_http_status(302)
-    #     expect(response).to redirect_to(root_path)
-    #   end
-    # end
+      expect(response).to be_success
+      expect(response).to have_http_status(200)
+    end
+    context "when an order is paid" do
+      it "renders the receipt template" do
+        get :receipt
+
+        expect(response).to render_template("receipt")
+      end
+
+      it "resets session[:order_id]" do
+        get :receipt
+
+        expect(session[:order_id]).to eq(nil)
+      end
+    end
+
+    context "when an order is not paid" do
+      let(:test_order_pending) { Order.create( status: "pending", buyer_name: "My name", buyer_email: "my_email@example.com", buyer_address: "123 Example St, Cityville, State 12345", buyer_card_short: "1234", buyer_card_expiration: future_date ) }
+      let(:test_order_complete) { Order.create( status: "complete", buyer_name: "My name", buyer_email: "my_email@example.com", buyer_address: "123 Example St, Cityville, State 12345", buyer_card_short: "1234", buyer_card_expiration: future_date ) }
+      let(:test_order_canceled) { Order.create( status: "canceled", buyer_name: "My name", buyer_email: "my_email@example.com", buyer_address: "123 Example St, Cityville, State 12345", buyer_card_short: "1234", buyer_card_expiration: future_date ) }
+
+
+      it "redirects to root when 'pending'" do
```

Comment:
It seems like these three tests would be nicely combined by putting them each in an array and then iterating over that and executing the same test for each item in the array.